### DISCUSSION TITLE
SimpleLogin.io needs better auditing and control options for domain admins

### DISCUSSION THREAD

**Discussion by WinkelCode** [simple-login/app#557] [Feature Requests]
### Background:

I own multiple domains which I use with various services such as CloudFlare, DigitalOcean and Microsoft 365. It is my duty as the administrator of these domains to monitor them and ensure they aren’t being misused for nefarious purposes, and should such a thing have happened, to intervene and regain control.
The services mentioned above offer auditing tools which log certain aspects of interaction with their service and thus the associated domains and hostnames.

For example: Cloudflare is logging virtually all interactions with their service in detail, including IP address, time and the exact action that was taken. In fact, they even log actions taken on the account by their automated services, mostly related to their issued certificates.

DigitalOcean is a bit less verbose but is still reliably logging logins, two factor authentications and changes made via the account.

Microsoft 365 may be the most relevant example I can give, given section C. of “Alias is the new email address” on the simplelogin.io home page which is advertising SimpleLogin as an alternative to Google Workspace. While I admittedly have never used Google Workspace, it does seem to over comparable features to Microsoft 365, particularly regarding mail logging (https://support.google.com/a/answer/2618874?hl=en).
Besides also offering extensive audit logs, here I can select a time up to 90 days ago and generate a CSV file with a list of the email traffic to and from the associated domains.

Unauthorized access to a service can happen to anyone, regardless of measures like 2 factor authentication. It’s one thing to prevent unauthorized access, but equally as important is being able to detect and resolve it, this is especially important in cases like this where I own the domain and thus carry a lot of the responsibility for what its associated services do with it.

A particularly concerning point to me is the implementation feature to “share” aliases on SimpleLogin, including those on custom domains. I tested the feature using a throwaway account and when I transferred the domain, I merely received an email notifying me of this and the alias was transferred. There was no indication on the control panel anyone else was now in control of a part of the domain associated email service, which was authorized to send emails on its behalf in the SPF record. ~~In fact, in the domain panel, the number of aliases associated with the domain was only the number that was on my account~~, would the email not arrived (something which can happen to anyone various reasons) I would have had no idea this was happening.

Furthermore, there was nothing that I could’ve done about it – not even an “undo” link in the email.

### ...that being said:

I really like SimpleLogin and even bought a subscription, being able to “just” create aliases, and even allow you to reply to emails using the reverse-alias system, is really neat! Separating login emails and just burning the alias if it starts receiving spam is incredibly useful. I just wish it offered more control about the custom domains used with the service.

Here are some ideas to improve SimpleLogin in that aspect:

-	Add a section for account auditing, it should maintain a log* of the following:

       Logins (username/password), successful 2fa challenges, failed logins, changes to authentication (changes to password, 2FA, email, etc.), alias creation/removal/enabling/disabling, changes to the receiving mailbox, changes related to domains (removal, addition), and of course alias transfers.

-	Add a section for domain management, it should contain the following controls:

       [Controls already exposed in the current control panel] + List of all aliases on that domain, regardless of which user has the alias, a brief overview of activity on the alias (# of emails received and # of emails sent in the last x days), the ability to immediately “recover” the alias and return it to the current account.

       Side note: There really isn’t any point in offering privacy controls to people who have an alias on someone else’s domain, the owner could always just point the MX record somewhere else and see everything anyway.

-	Add a “mail flow” section*, possibly for custom domains only.

       Allow the owner of a domain to download or view a brief overview (sending/receiving addresses, dates and email subject) of emails that are coming in and out of the domains associated aliases.

       This setting should not be affected by the removal of connected reverse aliases as is currently the case.

*A quick word about privacy:

A lot of people who are using this service are probably not too keen on having such extensive logs of their activities kept, and SimpleLogin does specifically talk about keeping a minimum of data.
However, there is a comparable service, possibly even more focused on privacy that manages to do both: ProtonMail.
I am also a customer of ProtonMail and they offer a nice compromise between audit options and privacy as they have an audit log that can be turned on or off by the user, and the logging of IP addresses can also be enabled or disabled separately, I believe that such an approach, of letting the user decide how much they want to have logged would offer a great compromise between the “no logs” and “Microsoft 365/Google Workspace-but-with-your-own-email” types.

I hope my feedback is considered for future development.

**Comment by nguyenkims** [comment]
Thanks a lot for the suggestions! The account auditing section is indeed necessary. About the "mail flow" section, I think it's more relevant for the enterprise tier in SimpleLogin. Btw we don't store the email subject so this section will consist mostly of sender address, timestamp, alias.

> Side note: There really isn’t any point in offering privacy controls to people who have an alias on someone else’s domain, the owner could always just point the MX record somewhere else and see everything anyway.

The alias transfer is designed to be used for giving a team member an alias. SimpleLogin is also used for managing small projects and the project administrator might want to give another member an alias that belongs to the project domain.

**Reply by WinkelCode** [reply]
Thank you for looking into my suggestions, I am looking forward to seeing the additions to account management implemented.

---

> >Side note: There really isn’t any point in offering privacy controls to people who have an alias on someone else’s domain, the owner could always just point the MX record somewhere else and see everything anyway.
> 
> The alias transfer is designed to be used for giving a team member an alias. SimpleLogin is also used for managing small projects and the project administrator might want to give another member an alias that belongs to the project domain.

I intended this comment to be about "retrieving" aliases which were transferred. Should such a feature be integrated, it should function differently than transferring aliases on SimpleLogin-owned domains, to my understanding currently the two systems are pretty much identical.

While a user who got an alias transferred to them on a SimpleLogin domain might expect to "own" that alias, on a custom domain, the domain's administrator can intercept emails by just changing the MX records, as such the actual ownership always remains with the domain admin who should be able to retrieve aliases instantly should they choose to.
